analytics_data_api/v0/tests/views/test_learners.py

```diff
@@ -0,0 +1,88 @@
+
+
+class LearnerTests(TestCaseWithAuthentication):
+    path_template = '/api/v0/learners/{}/?course_id={}'
+
+    @classmethod
+    def get_fixture(cls):
+        return {
+            "took": 11,
+            "_shards": {
+                "total": 10,
+                "successful": 10,
+                "failed": 0
+            },
+            "hits": {
+                "total": 33701,
+                "max_score": 7.201823,
+                "hits": [{
+                    "_index": "roster_1_1",
+                    "_type": "roster_entry",
+                    "_id": "edX/DemoX/Demo_Course|ed_xavier",
+                    "_score": 7.201823,
+                    "_source": {
+                        "username": "ed_xavier",
+                        "enrollment_mode": "honor",
+                        "problems_attempted": 43,
+                        "problems_completed": 3,
+                        "problem_attempts": 0,
+                        "course_id": "edX/DemoX/Demo_Course",
+                        "videos_viewed": 6,
+                        "name": "Edward Xavier",
+                        "segments": ["has_potential"],
+                        "email": "ed_xavier@example.com"
+                    }
+                }]
+            }
+        }
+
+    @classmethod
+    def setUpClass(cls):
+        # mock the elastic search client
+        client = Mock()
+        client.search.return_value = cls.get_fixture()
+        connections.add_connection('default', client)
+
+    def test_get_user(self):
+        user_name = 'ed_xavier'
+        course_id = 'edX/DemoX/Demo_Course'
+        response = self.authenticated_get(self.path_template.format(user_name, course_id))
+        self.assertEquals(response.status_code, 200)
+
+        expected = {
+            "username": "ed_xavier",
+            "enrollment_mode": "honor",
+            "name": "Edward Xavier",
+            "email": "ed_xavier@example.com",
+            "account_url": "http://lms-host/ed_xavier",
+            "segments": ["has_potential"],
+            "engagements": {
+                "problem_attempts": 0,
+                "problems_attempted": 43,
+                "problems_completed": 3,
+                "videos_viewed": 6,
+                "discussions_contributed": 0
+            },
+        }
+        self.assertDictEqual(expected, response.data)
+
+    @patch('analytics_data_api.v0.models.RosterEntry.get_course_user', Mock(return_value=[]))
+    def test_not_found(self):
+        user_name = 'a_user'
+        course_id = 'edX/DemoX/Demo_Course'
+        response = self.authenticated_get(self.path_template.format(user_name, course_id))
+        self.assertEquals(response.status_code, 404)
+
+    def test_get_400(self):
+        base_path = '/api/v0/learners/{}'
+        path = (base_path).format('ed_xavier')
+        response = self.authenticated_get(path)
+        self.assertEquals(response.status_code, 400)
```

Comment:
One final request: I'd prefer if the test cases are split up by edge cases rather than error codes. It makes the test cases more targeted and descriptive; we could have a `test_no_course_id`, and a `test_bad_course_id` rather than `test_get_400`, for example.

Also, now that we have awesome error handling, could you verify that the JSON corresponding to the error codes is correct?